src/main/java/com/manywho/services/email/email/decisions/clients/EngineClient.java

```diff
@@ -48,7 +49,18 @@ public String sendResponseToEngine(EmailDecisionRequest emailRequest, String str
                 flowState.sync();
             }
 
-            return runClient.join(emailRequest.getTenantId().toString(), flowState.getState().toString()).execute().body().getJoinFlowUri();
+            /**
+             * The sync() call can potentially clear out the invoke response that hangs off of the flowState
+             * instance (this seems to typically occur when there are no map elements to move on to). If an
+             * invoke response has been generated by the SDK, then grab the state ID from there, if not
+             * then just grab it from the incoming request.
+             * Being overly cautious here, but am unsure as to whether the a state ID might change when invoking
+             * the flows state, for instance with a flow that utilses a child flow (old subflows). So will continue
+             * to get the state ID from flowState if possible.
+             */
+            UUID stateId = flowState.getInvokeResponse() != null ? flowState.getState() : emailRequest.getStateId();
```

Comment:
@jamesbrattBoomi We don't have a style guide, but I think the inline condition is something that I am going to see more often in our projects.

Jonjo have suggested some changes that I think make sense:
```
UUID stateId = flowState.getInvokeResponse() == null
       ? emailRequest.getStateId()
       : flowState.getState();
```
No big difference but the condition is in positive, and using new lines.
Does it work for you?

Reply:
Yep, that's fine. Might be worth looking into introducing whatever Javas version of something like eslint is.